db/migrations/dedupe_github_repo_owner.py

```diff
@@ -0,0 +1,61 @@
+"""Remove case-sensitive duplicates in GitHub repository tables"""
+
+import rethinkdb as r
+
+import db.plugins
+import db.util
+
+r_conn = db.util.r_conn
+
+
+if __name__ == '__main__':
+    table = 'plugin_github_repos'
+
+    print 'Removing duplicate rows in %s' % table
+
+    updated = 0
+    deleted = 0
+
+    query = r.table(table)
+
+    # Group by the normalized GitHub path
+    query = query.group([
+        r.row['owner'].downcase(),
+        r.row['repo_name'].downcase()])
+
+    # Get the most recently scraped row in each group first
+    query = query.order_by(r.desc('last_scaped_at'))
+
+    grouped_repos = query.run(r_conn())
+
+    for owner_repo, repos in grouped_repos.iteritems():
+
+        print '\nRepo with GitHub path %s occurs %s times' % (
+                owner_repo,
+                len(repos))
+
+        # Use the most recently scraped row as the canonical row
+        canonical = repos.pop()
+
+        assert canonical
```

Comment:
This would potentially crash out after data has been modified in the database, though, correct? i.e. calling `run` in iterations prior to one during which an `AssertionError` is raised will have migrated some data (but not all that was intended). If this is the intended execution plan, there's nothing inherently wrong with it. My personal approach would be either to check for errors and bail out before anything gets modified or to migrate everything that can be and log the problematic records for further review. Just food for thought.